### PR TITLE
Add suppress_push_request_jira as editable field

### PR DESCRIPTION
**Pull request by CryptoRodeo**
Add the `suppress_push_request_jira` field for:
- product
- product_version

JIRA: CWFCONF-5478

### CODE REVIEW

**Comment by CryptoRodeo** [comment]
The integration failure seems to be an ET issue. I just noticed that this is now an allowable param yet...

**Comment by CryptoRodeo** [comment]
@ktdreyer PTAL. I was able to test this locally with the changes from `CWFCONF-5479` using the integration tests. I can confirm that the value `suppress_push_request_jira` was changed for both products and product versions. Also being able to specify `null` as a value worked when updating product versions.

File: library/errata_tool_product_version.py [hunk 1]
```diff
@@ -281,6 +295,7 @@ def run_module():
         brew_tags=dict(type='list', required=True),
         use_quay_for_containers=dict(type='bool'),
         use_quay_for_containers_stage=dict(type='bool'),
+        suppress_push_request_jira=dict(type='bool', default=None),
```
Comment (ktdreyer) [inline library/errata_tool_product_version.py:298]: Ansible already sets `default=None` for all parameters, so you can remove it here.

I had to learn this the hard way, and I went through the codebase a while back and purged all the `default=None` instances in 847dc6110f1b7817aed0c2a04c6d8eb032f2b8b2 and 654df3784d83f7dadc15fbac4057299a7bae4e98

File: library/errata_tool_product_version.py [hunk 2]
```diff
@@ -112,6 +112,16 @@
        - What are the consequences of a completely empty brew_tag list? This
          might be answered in CLOUDWF-2.
      required: true
+   suppress_push_request_jira:
+     description:
+       - Set to true to suppress creating push request jira tickets.
+       - Set to false to prevent push request jira tickets and override the
+         associated product's setting.
```
Comment (ktdreyer) [inline library/errata_tool_product_version.py:119]: `false` will *prevent* Jira tickets?

(This is a copy-paste from Gerrit 447193)

File: library/errata_tool_product_version.py [hunk 3]
```diff
@@ -242,6 +253,9 @@ def prepare_diff_data(before, after):
 def ensure_product_version(client, params, check_mode):
     result = {'changed': False, 'stdout_lines': []}
     params = {param: val for param, val in params.items() if val is not None}
+    # Edge case, this value is allowed to be None.
+    if 'suppress_push_request_jira' not in params:
+        params['suppress_push_request_jira'] = None
```
Comment (ktdreyer) [inline library/errata_tool_product_version.py:258]: With  `default=None`, all existing playbooks will start wiping the existing values out on the server.

We could pass a special constant like `omitted` here or something, but that's the first time I can remember seeing this in an ET REST API. This steers into some choppy edge-case waters.

For this PR, how about you just drop these two lines, and not make it possible to set this to `null` with Ansible yet? If it's really important to support this edge case, you can add it in another PR (with more testing).

**Review by ktdreyer — CHANGES_REQUESTED** [review]

File: library/errata_tool_product_version.py (hunk 1 above)
Comment (CryptoRodeo) [reply library/errata_tool_product_version.py]: Thanks for that insight, removed

File: library/errata_tool_product_version.py (hunk 2 above)
Comment (CryptoRodeo) [reply library/errata_tool_product_version.py:119]: Good catch! I've updated the doc

File: library/errata_tool_product_version.py (hunk 3 above)
Comment (CryptoRodeo) [reply library/errata_tool_product_version.py]: Good idea, I removed that edge case

**Comment by ktdreyer** [comment]
Looks like GitHub cannot upload this to galaxy.ansible.com now. I've filed https://github.com/ktdreyer/errata-tool-ansible/issues/308 to track that.

**Pull request merged**